src/core/credentials/transport/alts/alts_security_connector.cc

```diff
@@ -107,8 +110,12 @@ class grpc_alts_channel_security_connector final
               true, interested_parties, &handshaker,
               user_specified_max_frame_size,
               args.GetString(GRPC_ARG_TRANSPORT_PROTOCOLS)) == TSI_OK);
-    handshake_manager->Add(
-        grpc_core::SecurityHandshakerCreate(handshaker, this, args));
+    protocols_ = alts_tsi_handshaker_get_protocols_for_testing(handshaker);
```

Comment:
A method marked `for_testing` shouldn't be used in a regular codepath - is this needed, and if so we shouldn't mark it `for_testing`